### PR TITLE
add a few useful snippets

### PR DESCRIPTION
**Pull request by marklap**
Useful snippets to me that is.

### CODE REVIEW

File: snippets/go.json [hunk 1]
```diff
@@ -155,6 +155,18 @@
 		"test function": {
 			"prefix": "tf",
 			"body": "func Test$1(t *testing.T) {\n\t$0\n}"
+		},
+		"init function": {
+			"prefix": "finit",
```
Comment (kulshekhar) [inline snippets/go.json]: All things considered, `fmain` and `init` seem like the most apt options

Comment (marklap) [inline snippets/go.json]: `fmain` and `init` work for me.

Comment (marklap) [inline snippets/go.json]: Someone give me the nod and I'll update my PR to change `finit` to `init`.

Comment (bennyscetbun) [inline snippets/go.json]: Hi my thought on this one.
I like consistency and many dev does too.
main/init should be good except that the #225 is a real pain when using it
So fmain/finit works fine for me :)
But please. No fmain/init or main/finit :)

Comment (patbaker82) [inline snippets/go.json]: I agree with @bennyscetbun.

Comment (ramya-rao-a) [inline snippets/go.json]: Consistency does make sense and drive usability as well. Lets see if there is any other feedback over the weekend. I'll merge the PR (`fmain` and `finit`) on Monday if we don't hear anything more.

Comment (kulshekhar) [inline snippets/go.json]: Given that 'test function' uses `tf` and 'goroutine function' uses `gf`, maybe the main function can use `mf` while the 'init' function can use `init` (`if` might not be the best choice :slightly_smiling_face:)

Comment (ramya-rao-a) [inline snippets/go.json]: sigh.. This takes me back to a programmer's hardest task. http://imgur.com/y3dhJFQ : Naming things.

Going with `fmain` and `finit`

**Pull request merged**